compensations/src/main/java/org/jboss/narayana/compensations/impl/ParticipantImpl.java

```diff
@@ -92,9 +94,10 @@ public void close() throws WrongStateException, SystemException {
             ConfirmationHandler handler = instantiate(confirmationHandler);
             handler.confirm();
 
-            CompensationContext.close(currentTX);
             Thread.currentThread().setContextClassLoader(origClassLoader);
         }
+
+        CompensationContextLifetimeManager.getInstance().participantDestroyed(currentTX);
```

Comment:
Ah yes I know that. I wasn't sure about this one, because before context destruction was without try/catch as well. So if e.g. close operation fails context does not close.